packages/neon-core/src/u/HexString.ts

```diff
@@ -0,0 +1,80 @@
+import { ensureHex, reverseHex, hexXor } from "./misc";
+import {
+  str2hexstring,
+  num2hexstring,
+  hexstring2ab,
+  ab2hexstring,
+  hexstring2str
+} from "./convert";
+
+export class HexString {
+  /**
+   * Will store value in big endian
+   */
+  private _value: string;
+
+  protected _checkValue(value: string) {
+    ensureHex(value);
+  }
+
+  protected constructor(value: string, littleEndian = false) {
+    if (value.startsWith("0x")) {
+      value = value.slice(2);
+    }
+    this._checkValue(value);
+    this._value = littleEndian ? reverseHex(value) : value;
+  }
+
+  public toBigEndian(): string {
+    return this._value;
+  }
+
+  public toLittleEndian(): string {
+    return reverseHex(this._value);
+  }
+
+  public reverse(): this {
+    this._value = reverseHex(this._value);
+    return this;
+  }
+
+  public isEqualTo(other: HexString): boolean {
```

Comment:
we use `equals` on all our other classes so stick with the convention.

Reply:
OK